src/sidebar/components/test/search-input-test.js

```diff
@@ -1,68 +1,88 @@
 'use strict';
 
-const angular = require('angular');
+const { createElement } = require('preact');
+const { mount } = require('enzyme');
 
-const util = require('../../directive/test/util');
+const SearchInput = require('../search-input');
 
-describe('searchInput', function() {
+describe('SearchInput', () => {
   let fakeStore;
 
-  before(function() {
-    angular
-      .module('app', [])
-      .component('searchInput', require('../search-input'));
-  });
+  const createSearchInput = (props = {}) =>
+    // `mount` rendering is used so we can get access to DOM nodes.
+    mount(<SearchInput {...props} />);
+
+  function typeQuery(wrapper, query) {
+    const input = wrapper.find('input');
+    input.getDOMNode().value = query;
+    input.simulate('input');
+  }
 
-  beforeEach(function() {
+  beforeEach(() => {
     fakeStore = { isLoading: sinon.stub().returns(false) };
-    angular.mock.module('app', {
-      store: fakeStore,
+
+    const FakeSpinner = () => null;
+    FakeSpinner.displayName = 'Spinner';
+
+    SearchInput.$imports.$mock({
+      './spinner': FakeSpinner,
+      '../store/use-store': callback => callback(fakeStore),
     });
   });
 
-  it('displays the search query', function() {
-    const el = util.createDirective(document, 'searchInput', {
-      query: 'foo',
-    });
-    const input = el.find('input')[0];
-    assert.equal(input.value, 'foo');
+  afterEach(() => {
+    SearchInput.$imports.$restore();
   });
 
-  it('invokes #onSearch() when the query changes', function() {
-    const onSearch = sinon.stub();
-    const el = util.createDirective(document, 'searchInput', {
-      query: 'foo',
-      onSearch: {
-        args: ['$query'],
-        callback: onSearch,
-      },
-    });
-    const input = el.find('input')[0];
-    const form = el.find('form');
-    input.value = 'new-query';
-    form.submit();
-    assert.calledWith(onSearch, 'new-query');
+  it('displays the active query', () => {
+    const wrapper = createSearchInput({ query: 'foo' });
+    assert.equal(wrapper.find('input').prop('value'), 'foo');
   });
 
-  describe('loading indicator', function() {
-    it('is hidden when there are no API requests in flight', function() {
-      const el = util.createDirective(document, 'search-input', {});
-      const spinner = el[0].querySelector('spinner');
+  it('resets input field value to active query when active query changes', () => {
+    const wrapper = createSearchInput({ query: 'foo' });
 
-      fakeStore.isLoading.returns(false);
-      el.scope.$digest();
+    // Simulate user editing the pending query, but not committing it.
+    typeQuery(wrapper, 'pending-query');
 
-      assert.equal(util.isHidden(spinner), true);
-    });
+    // Check that the pending query is displayed.
+    assert.equal(wrapper.find('input').prop('value'), 'pending-query');
```

Comment:
Is it worth checking that the `query` value is unchanged at this point?

Reply:
The `query` value comes from above, so it won't change unless you do `wrapper.setProps({ query: someNewValue })`